includes/payment-methods/class-wc-stripe-express-checkout-ajax-handler.php

```diff
@@ -421,6 +422,20 @@ public function modify_country_locale_for_express_checkout( $locale ) {
 			}
 		}
 
+		// List of countries where postcode is optional in express checkouts (Google Pay, Apple Pay).
+		// These countries allow addresses without postal codes, but WooCommerce requires them by default.
+		$countries_with_optional_postcode = apply_filters(
+			'wc_stripe_express_checkout_countries_with_optional_postcode',
+			[
+				'IL', // Israel
+			]
+		);
+
+		// Make postcode optional for countries where payment providers don't require it.
+		foreach ( $countries_with_optional_postcode as $country_code ) {
+			$locale[ $country_code ]['postcode']['required'] = false;
```

Comment:
Is it worth being defensive to make sure the country code exists before we cause more problems?
```suggestion
			if ( isset( $locale[ $country_code ] ) ) {
				$locale[ $country_code ]['postcode']['required'] = false;
			}
```

Reply:
Good idea, it makes sense to add that check, given that we are allowing merchants to set the country_code in the filter above.